web_sdk/BUILD.gn

```diff
@@ -248,3 +251,157 @@ prebuilt_dart_action("flutter_dartdevc_canvaskit_kernel_sdk") {
         "$root_out_dir/flutter_web_sdk/kernel/legacy-canvaskit/dart_sdk.js"),
   ]
 }
+
+# Compiles the DDC CanvasKit SDK's JS code for null safety.
+prebuilt_dart_action("flutter_dartdevc_canvaskit_kernel_sdk_sound") {
+  deps = [
+    ":flutter_dartdevc_kernel_sdk_outline_sound",
```

Comment:
What are the outline dep for? I don't see it being passed as an arg.

In the SDK I use the full .dill to compile the js from .dill now. 

If you do need it, should this outline be the canvaskit version just to be safe? I suppose we want the API to be the same but is it a guaranty?

Reply:
The outlines are fed to the frontend_server for the hot reload cycle. We don't use them for building the precompiled SDK.

Their APIs should be the same since the only different is a define within the implementation.

Reply:
It seems like I can remove the dep though, since it isn't used

Reply:
What is the actual input given to DDC to compile the sdk? I don't see where we say use the sound vs unsound dill to compile the js.

Reply:
IIRC the source files are used directly, not the dills

Reply:
Thanks. I should say that part of my question is rooted in the investigation I'm doing to try and figure out how DDC is being used to compile the sdk in all the different environments. Flutter web is the one I understand the least at the moment.

Reply:
Happy to discuss in person more if it helps. The compilation needs should be fairly similar to whatever the dart SDK is doing, and the current choice of scripts/args is more than likely historical rather than intentional.